ovs/lib/mdsservice.py

```diff
@@ -156,92 +160,107 @@ def prepare_mds_service(storagerouter, vpool, fresh_only):
 
     @staticmethod
-    def remove_mds_service(mds_service, vpool, reconfigure, allow_offline=False):
+    def remove_mds_service(mds_service, reconfigure, allow_offline=False):
         """
         Removes an MDS service
         :param mds_service: The MDS service to remove
         :type mds_service: ovs.dal.hybrids.j_mdsservice.MDSService
-        :param vpool: The vPool for which the MDS service will be removed
-        :type vpool: ovs.dal.hybrids.vpool.VPool
         :param reconfigure: Indicates whether reconfiguration is required
         :type reconfigure: bool
         :param allow_offline: Indicates whether it's OK that the node for which mds services are cleaned is offline
         :type allow_offline: bool
+        :raises: RuntimeError - When vDisks present on the MDSService to be removed
+                              - No StorageDriver is linked to the MDSService to be removed
+                 UnableToConnectException - When StorageRouter on which the MDSService resides is unreachable and allow_offline flag is False
+        :return: None
+        :rtype: NoneType
         """
-        if mds_service.guid not in vpool.mds_services_guids:
-            raise RuntimeError('MDS Service {0} is not related to vPool {1}'.format(mds_service.service.name, vpool.name))
-
         if len(mds_service.vdisks_guids) > 0 and allow_offline is False:
             raise RuntimeError('Cannot remove MDSService that is still serving disks')
 
         if len(mds_service.storagedriver_partitions) == 0 or mds_service.storagedriver_partitions[0].storagedriver is None:
             raise RuntimeError('Failed to retrieve the linked StorageDriver to this MDS Service {0}'.format(mds_service.service.name))
 
-        # Clean up model
+        vpool = mds_service.vpool
+        root_client = None
         storagerouter = mds_service.service.storagerouter
         storagedriver = mds_service.storagedriver_partitions[0].storagedriver
+        MDSServiceController._logger.info('Remove MDS: StorageRouter {0} - vPool {1}: Removing MDS junction service for port {2}'.format(storagerouter.name, vpool.name, mds_service.service.ports[0]))
+        try:
+            root_client = SSHClient(endpoint=storagerouter, username='root')
+            MDSServiceController._logger.debug('Remove MDS: StorageRouter {0} - vPool {1}: Established SSH connection'.format(storagerouter.name, vpool.name))
+        except UnableToConnectException:
+            if allow_offline is True:
+                MDSServiceController._logger.warning('Remove MDS: StorageRouter {0} - vPool {1}: Allowed offline node during MDS service removal'.format(storagerouter.name, vpool.name))
+            else:
+                MDSServiceController._logger.exception('Remove MDS: StorageRouter {0} - vPool {1}: Failed to connect to StorageRouter'.format(storagerouter.name, vpool.name))
+                raise
+
+        # Reconfigure StorageDriver
+        if reconfigure is True and root_client is not None:
```

Comment:
Raise on reconfigure is True and root_client is None. The request action could not be done  so we should error here

Reply:
root_client will be None only when allow_offline is True. When allow_offline is True and the storagerouter is indeed offline, we can't reconfigure anyway.
If allow_offline would be False and the storagerouter cannot be reached, we raise in the else stack